test/org/sagebionetworks/bridge/upload/IosSchemaValidationHandler2Test.java

```diff
@@ -0,0 +1,418 @@
+                        .build(),
+                new DynamoUploadFieldDefinition.Builder().withName("baz")
+                        .withType(UploadFieldType.ATTACHMENT_JSON_BLOB).build()));
+
+        DynamoUploadSchema jsonDataSchema = new DynamoUploadSchema();
+        jsonDataSchema.setStudyId(TEST_STUDY_ID);
+        jsonDataSchema.setSchemaId("json-data");
+        jsonDataSchema.setRevision(1);
+        jsonDataSchema.setName("JSON Data");
+        jsonDataSchema.setSchemaType(UploadSchemaType.IOS_DATA);
+        jsonDataSchema.setFieldDefinitions(ImmutableList.<UploadFieldDefinition>of(
+                new DynamoUploadFieldDefinition.Builder().withName("string.json.string")
+                        .withType(UploadFieldType.STRING).build(),
+                new DynamoUploadFieldDefinition.Builder().withName("blob.json.blob")
+                        .withType(UploadFieldType.ATTACHMENT_JSON_BLOB).build()));
+
+        DynamoUploadSchema nonJsonDataSchema = new DynamoUploadSchema();
+        nonJsonDataSchema.setStudyId(TEST_STUDY_ID);
+        nonJsonDataSchema.setSchemaId("non-json-data");
+        nonJsonDataSchema.setRevision(1);
+        nonJsonDataSchema.setName("Non-JSON Data");
+        nonJsonDataSchema.setSchemaType(UploadSchemaType.IOS_DATA);
+        nonJsonDataSchema.setFieldDefinitions(ImmutableList.<UploadFieldDefinition>of(
+                new DynamoUploadFieldDefinition.Builder().withName("nonJsonFile.txt")
+                        .withType(UploadFieldType.ATTACHMENT_BLOB).build(),
+                new DynamoUploadFieldDefinition.Builder().withName("jsonFile.json")
+                        .withType(UploadFieldType.ATTACHMENT_JSON_BLOB).build()));
+
+        DynamoUploadSchema mixedSchema = new DynamoUploadSchema();
+        mixedSchema.setStudyId(TEST_STUDY_ID);
+        mixedSchema.setSchemaId("mixed-data");
+        mixedSchema.setRevision(1);
+        mixedSchema.setName("Mixed Data");
+        mixedSchema.setSchemaType(UploadSchemaType.IOS_DATA);
+        mixedSchema.setFieldDefinitions(ImmutableList.<UploadFieldDefinition>of(
+                new DynamoUploadFieldDefinition.Builder().withName("nonJsonFile.txt")
+                        .withType(UploadFieldType.ATTACHMENT_BLOB).build(),
+                new DynamoUploadFieldDefinition.Builder().withName("attachment.json")
+                        .withType(UploadFieldType.ATTACHMENT_JSON_BLOB).build(),
+                new DynamoUploadFieldDefinition.Builder().withName("inline.json")
+                        .withType(UploadFieldType.INLINE_JSON_BLOB).build(),
+                new DynamoUploadFieldDefinition.Builder().withName("field.json.attachment")
+                        .withType(UploadFieldType.ATTACHMENT_JSON_BLOB).build(),
+                new DynamoUploadFieldDefinition.Builder().withName("field.json.string")
+                        .withType(UploadFieldType.STRING).build()));
+
```

Comment:
I'm harping on this but:
mixedSchema.addFieldDefinition("nonJsonFile.txt", ATTACHMENT_BLOB);
mixedSchema.addFieldDefinition("attachment.json", ATTACHMENT_JSON);
mixedSchema.addFieldDefinition("inline.json", INLINE_JSON_BLOB);
mixedSchema.addFieldDefinition("field.json.attachment", ATTACHMENT_JSON_BLOB);
mixedSchema.addFieldDefinition("field.json.string", STRING);

Seems like a reasonable shortcut for these, assuming fieldDefinitions has a mutable list initialized. Could be made immutable when getFieldDefinitions() is called.